### PR TITLE
feature: generate rsa keys

### PR DESCRIPTION
**Pull request by spacewander**
Make it possible to generate rsa public key and private key on the fly.
I will add documentation in the next pr if this one is merged.

### CODE REVIEW

File: t/rsa.t [hunk 1]
```diff
@@ -582,3 +582,44 @@ true
 true
 --- no_error_log
 [error]
+
+
+=== TEST 11: generate RSA public key and private key
+--- http_config eval: $::HttpConfig
+--- config
+    location /t {
+        content_by_lua '
```
Comment (doujiang24) [inline t/rsa.t]: we'd better use `content_by_lua_block` for new test case :)

File: t/rsa.t [hunk 2]
```diff
@@ -582,3 +582,44 @@ true
...
+            $TEST_NGINX_PK_CONF
```
Comment (doujiang24) [inline t/rsa.t:593]: So, what is `TEST_NGINX_PK_CONF`? Am I missed something?

File: t/rsa.t [hunk 3]
```diff
@@ -582,3 +582,44 @@ true
 true
 --- no_error_log
 [error]
+
+
```
Comment (doujiang24) [inline t/rsa.t:586]: three blank lines needed here :)

**Review by doujiang24 — COMMENTED** [review]
@spacewander Nice job, thanks a lot :)

File: t/rsa.t (hunk 2 above)
Comment (doujiang24) [reply t/rsa.t:593]: Okay, I got it.

**Comment by doujiang24** [comment]
@spacewander I have invited you to be collaborators, feel free to push master, thanks :)

**Comment by spacewander** [comment]
@doujiang24 Thanks for your appreciation! I will update the test case and get it merged.

**Pull request merged**